docs/program/run.md

```diff
@@ -0,0 +1,81 @@
+---
+
+After saving your [code sample](/program/#hello-world-the-code-sample-tab) and adding control logic with [Viam's SDKs](/program/sdks/), run your program to control your Viam-connected robot.
+Do this by running a command to execute the program in the terminal of a machine with the appropriate programming language and Viam SDK installed.
+
+For example:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+python3 ~/myCode/myViamFile.py
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go run ~/myCode/myViamFile.py
+```
+
+{{% /tab %}}
+{{% tab name="TypeScript" %}}
+
+For an example, see [this execution demo.](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples/vanilla)
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+For information on running C++ code see [the instructions on GitHub](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+If you are connecting to your robot through an external signaling address, you can perform this step on any computer.
+
+This is useful because as long as your computer is able to establish a network connection with the robot's computer, your control logic will be executed on the robot.
+
+The code sample generated by the Viam app includes a Viam cloud address like [`12345.somerobot-main.viam.cloud`] as an external or public address to connect to your robot .
+Select **Include Secret** on the **Code sample** tab to make the code sample include the robot location secret, which is necessary to authenticate to the robot at that address.
+
+### Run Code Locally
+
+If you need to run [PID control loops](https://en.wikipedia.org/wiki/PID_controller) or other on-robot code, you can run control code on the same board that is running `viam-server`.
+
+In case your robots have intermittent internet connectivity, you can ensure this does not interfere with the code's execution, by following these steps:
+
+{{< alert title="Note" color="note" >}}
+Currently, this only works with Python code which is running on the same board that `viam-server` is running on.
+{{< /alert >}}
+
+1. Change the `connect()` method to disable {{< glossary_tooltip term_id="webrtc" >}} and add the auth_entity in the DialOptions and use `localhost:8080`:
+
+    ```python {class="line-numbers linkable-line-numbers" data-line="5"}
+    async def connect():
+      creds = Credentials(type='robot-location-secret', payload=PAYLOAD_SECRET)
+      opts = RobotClient.Options(
+        refresh_interval=0,
+        dial_options=DialOptions(
+            credentials=creds,
+            disable_webrtc=True,
+            auth_entity=ROBOT_NAME
+        )
+      )
+      return await RobotClient.at_address('localhost:8080', opts)
+    ```
+
+2. Replace the `ROBOT_NAME` with your robot's Viam cloud address and the `PAYLOAD_SECRET` with your robot secret.
```

Comment:
Do we need to cover this here as this is not specific to running code locally on-robot

Reply:
@mcvella This code is from Naomi's ticket moved over, but I agree it's now redundant. Actually taking a bit of time to look over this whole section now, will ping you when ready for re-review!

Reply:
Step three isn't specific to this? I thought changing to localhost is a requirement